accounts/keystore/keystore.go

```diff
@@ -228,6 +228,8 @@ func (ks *KeyStore) Accounts() []accounts.Account {
 // Delete deletes the key matched by account if the passphrase is correct.
 // If the account contains no filename, the address must match a unique key.
 func (ks *KeyStore) Delete(a accounts.Account, passphrase string) error {
+	ks.cache.fileC.mu.Lock()
+	defer ks.cache.fileC.mu.Unlock()
```

Comment:
This is a bit of a "code-smell", to me. The KeyStore just reaches right into the member `cache`, and then reaches  one step further into the `fileC` and just uses it's mutex.  
There must be some more modular way of achieving the goal (I'm not 100% sure what the underlying problem that this solves, is).

Reply:
One way to do it would be to first load and zero the key, and when that is done, invoke `ks.cache.Remove(a)`. And then `Remove` would obtain the lock, delete the file on disk, delete it in cache. 

So instead of 
```golang
	err = os.Remove(a.URL.Path)
	if err == nil {
		ks.cache.delete(a)
		ks.refreshWallets()
	}
	return err
```

```golang
	if err = ks.cache.DeleteFile(a); err != nil{
		return err
	}
	ks.refreshWallets()
	return nil
```
Just an idea...